server/server.go

```diff
@@ -835,14 +836,28 @@ func (server *Server) loadEntryPointConfig(entryPointName string, entryPoint *En
 	return rewrite, nil
 }
 
-func (server *Server) buildDefaultHTTPRouter() *mux.Router {
+func (server *Server) buildDefaultHTTPRouter(globalConfiguration GlobalConfiguration) *mux.Router {
 	router := mux.NewRouter()
-	router.NotFoundHandler = http.HandlerFunc(notFoundHandler)
+	router.NotFoundHandler = newNotFounderHandler(globalConfiguration)
 	router.StrictSlash(true)
 	router.SkipClean(true)
 	return router
 }
 
+func newNotFounderHandler(globalConfiguration GlobalConfiguration) http.Handler {
```

Comment:
I think we should only pass in `globalConfiguration.NoRouteResponseCode` here as the function does not care about the remaining global configuration parameters.